applications/nrf_desktop/doc/dfu_lock.rst

```diff
@@ -0,0 +1,52 @@
+.. _nrf_desktop_dfu_lock:
+
+DFU lock utility
+################
+
+.. contents::
+   :local:
+   :depth: 2
+
+The DFU lock utility provides a synchronization mechanism for accessing the DFU flash.
+It is needed for application configurations that support more than one DFU transports.
+
+The DFU lock utility provides a basic synchronization API for all declared DFU owners.
+Each DFU owner can be declared using the :c:struct:`dfu_lock_owner` structure.
+The :c:func:`dfu_lock_claim` function is used to claim ownership over the DFU flash.
+This function returns an error if the lock has already been taken by another DFU owner.
+The :c:func:`dfu_lock_release` function is used to free the DFU flash from the current owner.
+This function returns an error on the release attempt that is not triggered by the current owner.
+The :c:member:`dfu_lock_owner.owner_changed` callback is used to indicate the change in ownership.
+The previous owner can use this callback for tracking the DFU flash status and the need to erase it before subsequent DFU attempts.
+
+.. note::
+    The nRF Desktop DFU transports must voluntary take lock before accessing flash and release it after they stop using it.
+    The DFU lock utility does not provide any protection against the DFU transport module that writes to flash memory without taking a lock.
+
+Configuration
+*************
+
+Use the :ref:`CONFIG_DESKTOP_DFU_LOCK <config_desktop_app_options>` option to enable the utility module.
+This option is automatically enabled if more that one DFU transport is enabled in the application configuration.
+
+Currently, the DFU lock utility is automatically used if you enable the following DFU transports in your application:
```

Comment:
```suggestion
Currently, the DFU lock utility is automatically used if you enable the following supported DFU transports in your application:
```
Since the Kconfig options are mentioned below, it is good to point to them.